src/components/Button/tests/Button.test.tsx

```diff
@@ -434,4 +445,19 @@ describe('<Button />', () => {
       });
     });
   });
+
+  describe('stretchContent', () => {
+    it('sets variant class', () => {
+      const warningSpy = jest
```

Comment:
Perform mocking and reseting of mocks in a beforeEach / afterEach method rather than within the test block. Eg. https://github.com/Shopify/polaris-react/blob/3d2da225b7c484e446dd1522cab0db0dfbfdbb87/src/components/Modal/components/Dialog/tests/Dialog.test.tsx#L10-L16

This is because if the mockRestore is within the test block and an expectation fails then the mockResore() won't be called and that mock will potentially leak in other test cases, which can lead to hard to diagnose cascading errors - SomeOther.test.tsx failing because it is inadvertently using a mock that was set but not cleared up in Button.test.tsx is hecking confusing (I've been bit by this before, it's a great way to waste an afternoon)